src/SearchParameters/index.js

```diff
@@ -103,13 +105,19 @@ var SearchParameters = function( newParameters ) {
 
 SearchParameters.prototype = {
   constructor : SearchParameters,
-  clearRefinements : function clearRefinements() {
-    return this.mutateMe( function( m ) {
-      m.facetsRefinements = {};
-      m.facetsExcludes = {};
-      m.disjunctiveFacetsRefinements = {};
-      m.numericRefinements = {};
-    } );
+
+  /**
+   * Remove all refinements (disjunctive + conjunctive + excludes + numeric filters)
+   * @method
+   * @param {string} [name] - If given, name of the facet / attribute on which  we want to remove all refinements
+   * @return {AlgoliaSearchHelper}
+   */
+  clearRefinements : function clearRefinements( name ) {
+    return this
+      .clearNumericRefinements( name )
+      .clearFacetRefinements( name )
+      .clearExcludeRefinements( name )
+      .clearDisjunctiveFacetRefinements( name );
```

Comment:
Done in https://github.com/algolia/algoliasearch-helper-js/commit/137784b098703a35f8912690f8cac4971e70e2e7 :+1: